src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt

```diff
@@ -61,7 +62,15 @@ interface ParsingStrategy {
                 Float::class -> FloatStrategy()
                 Int::class -> IntStrategy()
                 Date::class -> DateStrategy()
-                else -> StringStrategy(searchSpecAnnotation)
+                else -> getStrategyWithSuperClass(fieldClass, searchSpecAnnotation)
```

Comment:
I don't really like that we have to check which strategy was chosen in 2 different methods...
Can we not find a way to include EnumStrategy in the original when case ? 
I prefer even to change the when case to a when without arguments as so :
```
when {
                fieldClass == Double::class -> DoubleStrategy() 
.....
                else -> StringStrategy(searchSpecAnnotation)
            }
```

Reply:
companion object {
        fun getStrategy(fieldClass: KClass<out Any>, searchSpecAnnotation: SearchSpec): ParsingStrategy {
            return when {
                fieldClass == Boolean::class -> BooleanStrategy()
                fieldClass == Double::class -> BooleanStrategy()
                fieldClass == Float::class -> FloatStrategy()
                fieldClass == Int::class -> IntStrategy()
                fieldClass.superclasses[0] == Enum::class -> EnumStrategy()
                else -> StringStrategy(searchSpecAnnotation)
            }
        }
    }